pandas/core/indexes/datetimes.py

```diff
@@ -1435,7 +1435,12 @@ def get_loc(self, key, method=None, tolerance=None):
             try:
                 stamp = Timestamp(key, tz=self.tz)
                 return Index.get_loc(self, stamp, method, tolerance)
-            except (KeyError, ValueError):
+            except KeyError:
+                raise KeyError(key)
+            except ValueError as e:
+                # ndarray tolerance size must match target index size
```

Comment:
what is this for?

Reply:
list like tolerance size mismatch raises a ValueError, but so does something else, so I differentiate here...definitely a better way to do this.